From b8ecb3f9fab3350ac9fbf2c0585932a804483c36 Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Wed, 2 Apr 2014 22:12:58 +0000
Subject: [PATCH] NET-530 input parameter of
 org.apache.commons.net.ftp.FTP.__getReply(boolean) is not used

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/net/trunk@1584204 13f79535-47bb-0310-9956-ffa450edef68
---
 src/changes/changes.xml                           | 3 +++
 src/main/java/org/apache/commons/net/ftp/FTP.java | 4 +++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index 301f699ee..ca6887b93 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -65,6 +65,9 @@ The <action> type attribute can be add,update,fix,remove.
         <release version="3.4" date="2014-01-??" description="
 This is mainly a bug-fix release. See further details below.
         ">
+            <action issue="NET-530" type="fix" dev="sebb" due-to="fish ship">
+            input parameter of org.apache.commons.net.ftp.FTP.__getReply(boolean) is not used
+            </action>
             <action issue="NET-529" type="fix" dev="sebb" due-to="Putinas Piliponis">
             SubnetUtils throws exception on valid input
             </action>
diff --git a/src/main/java/org/apache/commons/net/ftp/FTP.java b/src/main/java/org/apache/commons/net/ftp/FTP.java
index 12e363f0e..0b639107c 100644
--- a/src/main/java/org/apache/commons/net/ftp/FTP.java
+++ b/src/main/java/org/apache/commons/net/ftp/FTP.java
@@ -361,7 +361,9 @@ private void __getReply(boolean reportReply) throws IOException
             while ( isStrictMultilineParsing() ? __strictCheck(line, code) : __lenientCheck(line));
         }
 
-        fireReplyReceived(_replyCode, getReplyString());
+        if (reportReply) {
+            fireReplyReceived(_replyCode, getReplyString());
+        }
 
         if (_replyCode == FTPReply.SERVICE_NOT_AVAILABLE) {
             throw new FTPConnectionClosedException("FTP response 421 received.  Server closed connection.");
